src/types/dict.rs

```diff
@@ -357,6 +366,24 @@ impl<'py> PyDictMethods<'py> for Bound<'py, PyDict> {
         BoundDictIterator::new(self.clone())
     }
 
+    #[cfg(all(Py_GIL_DISABLED, not(feature = "nightly")))]
```

Comment:
```suggestion
    #[cfg(not(feature = "nightly"))]
```

The critical section macros are a no-op on the GIL-enabled build, so I think allowing this function to work (and rely on the GIL for locking) makes it easier for users to adopt this function.

Reply:
Agreed, I also think this should be available on all builds, for simplicity.